### PR TITLE
API Journalist can add images to article 

### PR DESCRIPTION
**Pull request by kermit-klein**
[PT LINK](https://www.pivotaltracker.com/story/show/172999946)

- Add image to article post request tests
- Add ActiveStorage
- Add DecodeService
- Add attach_image method to article controller
- Add update article model to include image
- Add AWS gem and change ```development.rb``` and ```production.rb``` to include amazon
- Add test host to ```test.rb```
- Add amazon ```storage.yml```

### CODE REVIEW

**Review by ViaMarcus — COMMENTED** [review]
Did you try to solve the issue with special code for tests, loading a file in the test properly?

**Comment by kermit-klein** [comment]
> Did you try to solve the issue with special code for tests, loading a file in the test properly?

yes assets folder includes a jpg

**Review by designerofthing — APPROVED** [review]

File: app/serializers/article/index_serializer.rb [hunk 1]
```diff
@@ -1,9 +1,20 @@
 # frozen_string_literal: true
 
 class Article::IndexSerializer < ActiveModel::Serializer
-  attributes :id, :title, :category, :published_at
+  include Rails.application.routes.url_helpers
```
Comment (oliverochman) [inline app/serializers/article/index_serializer.rb]: out of scope

File: app/serializers/article/show_serializer.rb [hunk 2]
```diff
@@ -1,5 +1,16 @@
 # frozen_string_literal: true
 
 class Article::ShowSerializer < ActiveModel::Serializer
-  attributes :id, :title, :body
+  include Rails.application.routes.url_helpers
```
Comment (oliverochman) [inline app/serializers/article/show_serializer.rb]: out of scope

File: config/storage.yml [hunk 3]
```diff
@@ -2,6 +2,13 @@ test:
   service: Disk
   root: <%= Rails.root.join("tmp/storage") %>
 
-local:
-  service: Disk
-  root: <%= Rails.root.join("storage") %>
+#local:
```
Comment (oliverochman) [inline config/storage.yml]: no need to be commented out

File: spec/factories/articles.rb [hunk 4]
```diff
@@ -5,5 +5,8 @@
     title { 'this is title' }
     body { 'this is body' }
     category { 'sport' }
+    trait :with_image do
```
Comment (oliverochman) [inline spec/factories/articles.rb]: out of scope

File: spec/rails_helper.rb [hunk 5]
```diff
@@ -25,3 +25,7 @@
   config.include ResponseJSON
   config.filter_gems_from_backtrace('rack', 'railties', 'bootsnap')
 end
+
+FactoryBot::SyntaxRunner.class_eval do
```
Comment (oliverochman) [inline spec/rails_helper.rb]: out of scope

File: spec/requests/api/article_create_request_spec.rb [hunk 6]
```diff
@@ -39,7 +50,7 @@
 
   describe 'creation without article sets default value and' do
     before do
-      post '/api/articles', headers: headers, params: { title: 'A title', body: 'The body' }
+      post '/api/articles', headers: headers, params: { title: 'A title', body: 'The body', image: image }
     end
 
     it 'has a 200 response' do
```
Comment (oliverochman) [inline spec/requests/api/article_create_request_spec.rb:56]: need to see a sad path if there is no image

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: app/serializers/article/index_serializer.rb (hunk 1 above)
Comment (kermit-klein) [reply app/serializers/article/index_serializer.rb]: at which point should we add these ?

File: app/serializers/article/show_serializer.rb (hunk 2 above)
Comment (kermit-klein) [reply app/serializers/article/show_serializer.rb]: at which point should we add these ?

File: config/storage.yml (hunk 3 above)
Comment (kermit-klein) [reply config/storage.yml]: at which point should we add these ?

File: spec/factories/articles.rb (hunk 4 above)
Comment (kermit-klein) [reply spec/factories/articles.rb]: at which point should we add these ?

File: spec/rails_helper.rb (hunk 5 above)
Comment (kermit-klein) [reply spec/rails_helper.rb]: at which point should we add these ?

**Pull request merged**